lib/ts/recipeModule.ts

```diff
@@ -56,18 +58,28 @@ export default abstract class RecipeModule {
             remainingPath = new NormalisedURLPath(match[2]);
         }
 
+        const mtRecipe = MultitenancyRecipe.getInstanceOrThrowError();
+
         for (let i = 0; i < apisHandled.length; i++) {
             let currAPI = apisHandled[i];
             if (!currAPI.disabled && currAPI.method === method) {
                 if (this.appInfo.apiBasePath.appendPath(currAPI.pathWithoutApiBasePath).equals(path)) {
-                    return { id: currAPI.id, tenantId: DEFAULT_TENANT_ID };
+                    const finalTenantId = mtRecipe.recipeInterfaceImpl.getTenantId({
+                        tenantIdFromFrontend: DEFAULT_TENANT_ID,
+                        userContext,
+                    });
+                    return { id: currAPI.id, tenantId: finalTenantId === undefined ? DEFAULT_TENANT_ID : tenantId };
                 } else if (
                     remainingPath !== undefined &&
                     this.appInfo.apiBasePath
                         .appendPath(currAPI.pathWithoutApiBasePath)
                         .equals(this.appInfo.apiBasePath.appendPath(remainingPath))
                 ) {
-                    return { id: currAPI.id, tenantId };
+                    const finalTenantId = mtRecipe.recipeInterfaceImpl.getTenantId({
```

Comment:
make the input and output type of getTenantId non optional

Reply:
Done